File: SEM5_CODE/PYTHON/BASIC/17_palindrome.py
text = input("Enter a string: ")             # Read the string
reverse_text = ""                            # Store the reversed string

for character in text:                       # Read every character
    reverse_text = character + reverse_text  # Add the character at the beginning

if text == reverse_text:                     # Compare original and reversed strings
    print("Palindrome")                      # Display palindrome result
else:
    print("Not a palindrome")                # Display non-palindrome result

File: SEM3_CODE/PHP/BASIC/17_palindrome.php
<?php
$text = readline('Enter a string: ');                      // Read the string
$reverseText = '';                                         // Store the reversed string

for ($index = strlen($text) - 1; $index >= 0; $index--) {  // Read from the last character
    $reverseText = $reverseText . $text[$index];           // Add the current character
}

if ($text == $reverseText) {                               // Compare original and reversed strings
    echo 'Palindrome';                                     // Display palindrome result
} else {
    echo 'Not a palindrome';                               // Display non-palindrome result
}
?>
